lib/perl/Genome/Model/Tools/Picard/FilterSamReads.pm

```diff
@@ -53,51 +58,18 @@ sub help_detail {
 EOS
 }
 
-sub execute {
-    my $self = shift;
-
-    my $input_file = $self->input_file;
-    my $output_file = $self->output_file;
-
-    my $jar_path = $self->picard_path . '/FilterSamReads.jar';
-    unless ( -e $jar_path ) {
-        die( "Failed to find $jar_path!" .
-          "This command may not be available in version " . $self->use_version );
-    }
+sub minimum_version_required { '1.77'; }
+sub _jar_name { 'FilterSamReads.jar'; }
+sub _java_class { 'net.sf.picard.sam.FilterSamReads'; }
```

Comment:
GMT::Picard::Base takes care of adding the `net.sf` when it is necessary.  The `_java_class` is documented to expect return a list, which the class takes care of joining with `.`s as well.  (The current implementation still works if this is already a joined string, however.)